precompiles/outposts/osmosis/setup_test.go

```diff
@@ -19,6 +19,8 @@ import (
 const (
 	portID    = "transfer"
 	channelID = "channel-0"
+	// XCSContract placeholder until the XCS contract is deployed on the Osmosis test chain
+	XCSContract = "placeholder"
```

Comment:
this needs to be passed to the precompile on initialization

Reply:
After discussing we are moving away from this approach @fedekunze . To allow future dApp developers to use the Outpost and initialise their own XCS contract we will pass it into the `swap` function. The placeholder there is for tests only.

Reply:
> To allow future dApp developers to use the Outpost and initialise their own XCS contract we will pass it into the swap function

No this is not ideal, if they want to create their own swap logic with a custom XCS contract they can use the ICS20 Extension. The idea is for us to abstract that functionality altogether.

Also, it could be a risk because the user does not know if the XCS contract is the right one, so if the UI changes the address of the XCS it could steal users funds

Reply:
Ok makes sense, in that case since `reciever` on ICS20 and `contractAddr` have to be pointing to the XCS contract we can remove the param from the `swap` and make it simpler.